qemu/tests/qemu_guest_agent.py

```diff
@@ -160,6 +182,8 @@ def gagent_create(self, params, vm, *args):
     def setup_gagent_in_guest(self, params, vm):
         error.context("Setup guest agent in VM '%s'" % vm.name)
         self.gagent_install(params, vm, *[params.get("gagent_install_cmd")])
+        if params["os_type"] == "windows":
+            self.gagent_check_install_over(params, vm)
```

Comment:
You can overwrite the method `gagent_install` in class `QemuGuestAgentBasicCheckWin` to avoid handling windows thing in the base class.
```python
class QemuGuestAgentBasicCheckWin(QemuGuestAgentBasicCheck):

    @error.context_aware
    def gagent_install(self, params, vm, *args):
        super(QemuGuestAgentBasicCheckWin, self).gagent_install(self, params, vm, *args)
        # Moving things from *gagent_check_install_over* to here
        ...
```